pkg/lib/flavor/types/linux_platform_flavor.go

```diff
@@ -423,13 +415,111 @@ func (rhelpf LinuxPlatformFlavor) getDefaultMeasurement() ([]string, error) {
+
+	var pcrCollection []hcTypes.PCRS
+
+	// pull out the logs for the required PCRs from both banks
+	for pcr, rules := range pcrList {
+
+		pI := hcTypes.PcrIndex(pcr.Index)
+		var pcrInfo *hcTypes.Pcr
+		pcrInfo, _ = pcrManifest.GetPcrValue(hcTypes.SHAAlgorithm(pcr.Bank), pI)
+
+		if pcrInfo != nil {
+
+			var currPcrEx hcTypes.PCRS
+
+			currPcrEx.PCR.Index = pcr.Index
+			currPcrEx.PCR.Bank = pcr.Bank
+			currPcrEx.Measurement = pcrInfo.Value
+			currPcrEx.PCRMatches = true
+
+			// Populate Value
+			// Event logs if allowed
+			if includeEventLog {
+				var eventLogEqualEvents []hcTypes.EventLogCriteria
+				manifestPcrEventLogs, err := pcrManifest.GetPcrEventLogNew(hcTypes.SHAAlgorithm(pcr.Bank), pI)
+
+				// check if returned logset from PCR is nil
+				if manifestPcrEventLogs != nil && err == nil {
+
+					// Convert EventLog to flavor format
+					for _, manifestEventLog := range manifestPcrEventLogs {
+						if len(manifestEventLog.Tags) == 0 {
+							if rules.PcrEquals.IsPcrEquals {
+								eventLogEqualEvents = append(eventLogEqualEvents, manifestEventLog)
+							}
+						}
+						for _, tag := range manifestEventLog.Tags {
+							if _, ok := rules.PcrIncludes[tag]; ok {
+								currPcrEx.EventlogIncludes = append(currPcrEx.EventlogIncludes, manifestEventLog)
+							} else if rules.PcrEquals.IsPcrEquals {
+								if _, ok := rules.PcrEquals.ExcludingTags[tag]; !ok {
+									eventLogEqualEvents = append(eventLogEqualEvents, manifestEventLog)
+								}
+							}
+						}
+					}
+					if rules.PcrEquals.IsPcrEquals {
+						var EventLogExcludes []string
+						for excludeTag, _ := range rules.PcrEquals.ExcludingTags {
+							EventLogExcludes = append(EventLogExcludes, excludeTag)
+						}
+						currPcrEx.EventlogEqual = &hcTypes.EventLogEqual{
+							Events:      eventLogEqualEvents,
+							ExcludeTags: EventLogExcludes,
+						}
+					}
+				}
+			}
+
+			pcrCollection = append(pcrCollection, currPcrEx)
+		}
+	}
+	// return map for flavor to use
+	return pcrCollection
+}
+
+// UpdateMetaSectionDetails This method is used to update the meta section in flavor part
+func UpdateMetaSectionDetails(flavorPart cf.FlavorPart, newMeta *cm.Meta, flavorTemplates []hvs.FlavorTemplate) *cm.Meta {
+	log.Trace("flavor/types/linux_platform_flavor:UpdateMetaSectionDetails() Entering")
+	defer log.Trace("flavor/types/linux_platform_flavor:UpdateMetaSectionDetails() Leaving")
+
+	var flavorTemplateID []uuid.UUID
+	for _, flavorTemplate := range flavorTemplates {
+		flavorTemplateID = append(flavorTemplateID, flavorTemplate.ID)
+		var flavor *hvs.FlavorPart
+		switch flavorPart {
+		case cf.FlavorPartPlatform:
+			flavor = flavorTemplate.FlavorParts.Platform
+		case cf.FlavorPartOs:
+			flavor = flavorTemplate.FlavorParts.OS
+		case cf.FlavorPartHostUnique:
+			flavor = flavorTemplate.FlavorParts.HostUnique
+		}
+
+		if flavor != nil {
+			newMeta.Description["flavor_template_ids"] = flavorTemplateID
```

Comment:
this gets rewritten till the end of the loop, can this value be assigned outside of the loop?